plugin-core/src/main/java/appland/index/IndexUtil.java

```diff
@@ -1,17 +1,90 @@
 package appland.index;
 
+import appland.config.AppMapConfigFile;
+import appland.files.AppMapFiles;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.module.ModuleManager;
+import com.intellij.openapi.progress.ProcessCanceledException;
 import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.openapi.vfs.VfsUtilCore;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.util.concurrency.annotations.RequiresReadLock;
 import com.intellij.util.indexing.FileBasedIndex;
+import com.intellij.util.indexing.FileBasedIndexExtension;
 import com.intellij.util.indexing.ID;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 final class IndexUtil {
     // base version for our indexes, increase when the data structures or the parsing logic change
     static final int BASE_VERSION = 57;
 
+    // filenames covered by our own indexes
+    static final Set<String> indexedFilenames = Collections.unmodifiableSet(findIndexedFilenames());
+
+    /**
+     * @param project Current project
+     * @return Directories on the local filesystem, which contain indexable AppMap data.
+     * The directories will be used for watched roots and for index decisions of the IDE.
+     */
+    static Set<VirtualFile> findAppMapIndexDirectories(@NotNull Project project) {
+        ApplicationManager.getApplication().assertReadAccessAllowed();
+
+        var appMapDirs = VfsUtilCore.createCompactVirtualFileSet();
+        collectAppMapOutputDirectories(project, appMapDirs);
+        return appMapDirs;
+    }
+
+    private static void collectAppMapOutputDirectories(@NotNull Project project, @NotNull Set<VirtualFile> target) {
+        var configFiles = findConfigFilesByExcludedDirs(project);
+        for (var configFile : configFiles) {
+            var config = AppMapConfigFile.parseConfigFile(configFile);
+            if (config != null && config.getAppMapDir() != null) {
+                var appMapDir = VfsUtilCore.findRelativeFile(config.getAppMapDir(), configFile.getParent());
+                if (appMapDir != null && appMapDir.isValid() && appMapDir.isInLocalFileSystem()) {
+                    target.add(appMapDir);
+                }
+            }
+        }
+    }
+
+    /**
+     * Locate appmap.yml files without index access.
+     */
+    private static @NotNull Collection<VirtualFile> findConfigFilesByExcludedDirs(@NotNull Project project) {
+        var configFiles = VfsUtilCore.createCompactVirtualFileSet();
+        for (var module : ModuleManager.getInstance(project).getModules()) {
+            for (var root : ModuleRootManager.getInstance(module).getExcludeRoots()) {
```

Comment:
Typically the `appmap.yml` will be located in the root of the project or a sub-project or sub-module. The path should typically not be git ignored. Maybe I'm misunderstanding, but what drives the need to locate an `appmap.yml` from the parent(s) of an excluded path?

Reply:
An excluded folder here is the exclusion by the IDE, which is not the same as .gitignore. Typically, build folders are excluded, e.g. „module-root/target“. This method is locating appmap.yml files without index access by using the excluded folders as starting point. But after thinking about this, this code is a bit stupid. I‘ll update this to look at content roots instead (i.e. module-root instead of „module-root/target“) as starting point.

Reply:
I pushed https://github.com/getappmap/appmap-intellij-plugin/pull/347/commits/04498a1be13638e9cf314b0786f2a27746c09a7a to fix this.